### ISSUE TITLE
Add ability to support Location not via custom blocks

### ISSUE THREAD

**Issue by sergey-ganchuk** [puppetlabs/puppetlabs-apache#2410]
## Use Case
I was trying to configure reverse proxy and found that there is not possible to create location directives not via custom blocks.

## Describe the Solution You Would Like
Add ability to manage locations via puppet and hiera directly.

## Describe Alternatives You've Considered
Templating of full vhost configuration is not so good solution.

**Comment by ekohl** [comment]
Have you tried the `$apache::vhost::directories` parameter? That can be used, though it is fairly complex and you need to read the [template](https://github.com/puppetlabs/puppetlabs-apache/blob/main/templates/vhost/_directories.erb) itself to figure it out.

You can also use [`apache::vhost::proxy`](https://github.com/puppetlabs/puppetlabs-apache/blob/main/manifests/vhost/proxy.pp).

**Comment by sergey-ganchuk** [comment]
Yep, It would be more clear if there would be ability to use location directive either.

**Comment by ekohl** [comment]
You can, though it's not clearly documented with an example. The `provider` key can be used. What I pass elsewhere:
```puppet
$directories = [
  {
    'provider' => 'location',
    'path' => '/my/path',
    'request_headers' => ["set SSL_CLIENT_S_DN \"admin\""],
    'requires' => ["expr %{SSL_CLIENT_S_DN_CN} == \"${somevar}\""],
  },
]
```

**Comment by Ramesh7** [comment]
Thank you, @ekohl, for your contributions. @sergey-ganchuk, we haven't received any feedback from you yet. Could you please review and confirm if it is functioning as anticipated for you?
